src/fides/api/ctl/database/samples.py

```diff
@@ -0,0 +1,151 @@
+    for both database & SaaS connections. Extends the existing models with an
+    optional 'dataset' field.
+    """
+
+    dataset: Optional[FidesKey] = None
+    """
+    (Optional) Valid fides_key for an existing dataset that should be used
+    for this ConnectionConfig. If a dataset does not exist for this key,
+    creation will fail.
+    """
+
+
+def load_sample_connections_from_project() -> List[SampleConnection]:
+    """
+    Loads all the sample connection YAML files from the sample project by
+    traversing through the sample_connections/ folder, inspecting each file, and
+    parsing them into "ConnectionConfig" models that would be suitable for the
+    `PATCH /api/v1/connection` API.
+
+    While parsing the YAML file, this also performs ENV variable expansion,
+    similar to what Docker/Bash/etc. support. This allows secrets to be provided
+    as ENV var names (e.g. $FIDES_DEPLOY__CONNECTORS__POSTGRES__PASSWORD) and
+    replaced with "real" values at runtime.
+
+    After parsing the models, this function then *excludes* any connection missing
+    any of it's secrets. This prevents gotchas with failing connections at runtime
+    and also makes it easy to programmatically control which connections should be
+    enabled by simply adding/removing the secrets to the config file.
+
+    Returns a list of models, ready to be upserted!
+
+    NOTE: This function does quite a bit of magic that the CLI/API/etc. don't
+    support today, so it is likely to fall out of sync or break surprisingly.
+    We've added extra tests that are designed to be defensive and warn us if
+    something drifts.
+    """
+    sample_connections_path = files(fides.data.sample_project).joinpath(
+        "sample_connections/"
+    )
+    sample_connections = []
+    for yaml_path in sample_connections_path.iterdir():
+        with yaml_path.open("r") as file:
+            # Expand ENV vars when reading the YAML, to handle secrets
+            yaml_dict = load_sample_yaml_file(file, expand_vars=True)
+            connections = yaml_dict.get("connection", [])
+            sample_connections.extend(
+                [SampleConnection.parse_obj(e) for e in connections]
+            )
+
+    # Disable any connections whose "secrets" dict has empty values
+    for connection in sample_connections:
+        # If there are no secrets at all, disable!
+        if not connection.secrets:
+            connection.disabled = True
+            continue
+
+        # If any of the secret values are missing, disable!
+        for _, value in dict(connection.secrets).items():
+            if not value or value == "":
+                connection.disabled = True
+
+    # Exclude any disabled connections from the final results
+    return [e for e in sample_connections if not e.disabled]
+
+
+def load_sample_yaml_file(file: TextIO, expand_vars: bool = True) -> Dict:
+    yaml_str = file.read()
+    if expand_vars:
+        return yaml.safe_load(expandvars(yaml_str))
```

Comment:
Note the usage of `expandvars` here! It's pretty handy, it just zooms through the whole string and replaces `$FOO` with the value of `os.environs["FOO"]` 👍